### PR TITLE
Add pre-trained Roberta encoder for base and large architecture

### PR DESCRIPTION
**Pull request by Nayef211**
## Summary
- Adding pre-trained encoder for Roberta base and large architectures
- Added model integration tests to `test/integration_tests` folder 
- Added documentation describing roberta, the datasets it was trained on, and relevant license information

## Test
`pytest test/integration_tests/test_models.py`

## Followup

- [ ] Refactor the XLMR unit tests to integration tests similar to what is done in this PR
- [ ] Update integration tests to use pytest parameterization 
- [ ] Configure the integration tests to run periodically instead of on every PR
- [ ] Add bibliography to torchtext docs similar to [torchaudio](https://github.com/pytorch/audio/blob/main/docs/source/refs.bib)

### CODE REVIEW

**Review by parmeet — COMMENTED** [review]
Thanks @Nayef211, Overall LGTM! Couple of things, could you please update the models doc https://github.com/pytorch/text/blob/main/docs/source/models.rst. Also I am not sure why I am seeing some older code in bundler.py, could you try rebasing it on to latest main.

File: torchtext/models/roberta/bundler.py [hunk 1]
```diff
@@ -192,3 +192,61 @@ def encoderConf(self) -> RobertaEncoderConf:
...
+        ),
+        T.Truncate(510),
+        T.AddToken(token=0, begin=True),
+        T.AddToken(token=2, begin=False),
+    ),
+)
+
+ROBERTA_LARGE_ENCODER.__doc__ = (
+    '''
+    Roberta Encoder with Large configuration
```
Comment (mthrok) [inline torchtext/models/roberta/bundler.py:248]: I recommend to mention what dataset was used to train this model and copy-right information.

https://pytorch.org/audio/stable/pipelines.html#torchaudio.pipelines.WAV2VEC2_ASR_BASE_10M

Comment (parmeet) [reply torchtext/models/roberta/bundler.py:248]: cc: @kartikayk

File: torchtext/models/roberta/bundler.py [hunk 2]
```diff
@@ -192,3 +192,61 @@ def encoderConf(self) -> RobertaEncoderConf:
...
+    _encoder_conf=RobertaEncoderConf(vocab_size=50265),
+    transform=lambda: T.Sequential(
+        T.GPT2BPETokenizer(
+            encoder_json_path=urljoin(_TEXT_BUCKET, "gpt2_bpe_encoder.json"),
+            vocab_bpe_path=urljoin(_TEXT_BUCKET, "gpt2_bpe_vocab.bpe"),
+        ),
+        T.VocabTransform(
+            load_state_dict_from_url(urljoin(_TEXT_BUCKET, "roberta.vocab.pt"))
+        ),
+        T.Truncate(510),
```
[the remark this answers is not in the document]
Comment (Nayef211) [reply torchtext/models/roberta/bundler.py]: Done!

File: torchtext/models/roberta/bundler.py (hunk 1 above)
Comment (Nayef211) [reply torchtext/models/roberta/bundler.py:258]: @mthrok just added the model description from fairseq repo. I also put adding citations to torchtext docs as a followup item. Lmk if this looks okay

Comment (mthrok) [reply torchtext/models/roberta/bundler.py:258]: I think it's worth mentioning the names of the datasets. Reading the README and the paper, it seems that they used `BOOKCORPUS`, `CC-NEWS`, `OPENWEBTEXT` and `STORIES` for pre-training. Can you double-check?

Comment (mthrok) [reply torchtext/models/roberta/bundler.py:258]: > I also put adding citations to torchtext docs as a followup item.

Sounds good. use of `sphinx.bibtex` extension makes this easy.

https://github.com/pytorch/audio/blob/main/docs/source/refs.bib

Comment (Nayef211) [reply torchtext/models/roberta/bundler.py:262]: > I think it's worth mentioning the names of the datasets. Reading the README and the paper, it seems that they used `BOOKCORPUS`, `CC-NEWS`, `OPENWEBTEXT` and `STORIES` for pre-training. Can you double-check?

I just checked again and you are correct, those are the datasets that Roberta is trained on. I just added a paragraph on this in the documentation.

Comment (mthrok) [reply torchtext/models/roberta/bundler.py:262]: LGTM.

File: test/integration_tests/test_models.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import torch
+import torchtext
+
+from ..common.assets import get_asset_path
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+class TestModels(TorchtextTestCase):
+    def test_roberta_base(self):
```
Comment (mthrok) [inline test/integration_tests/test_models.py:9]: I think this can be written in much compact manner with parameterization.

Comment (Nayef211) [reply test/integration_tests/test_models.py:9]: Added a followup item. Let me do this in a separate PR

**Pull request merged**